src/navigation/RootStackNavigation.tsx

```diff
@@ -105,8 +108,9 @@ const RootStackNavigation = () => {
               <AnalyticsProvider>
                 <WalletConnectContextProvider>
                   <RootStack.Navigator initialRouteName="LandingScreen">
-                    {/* Sub-navigation with custom header */}
-                    <RootStack.Group screenOptions={{ headerTransparent: true }}>
+                    {/* Sub-navigation with custom header. Showing the header only when authenticated fixes the
+                    reanimated bug while still allowing animated transitions between screens */}
+                    <RootStack.Group screenOptions={{ headerTransparent: true, headerShown: isAuthenticated }}>
```

Comment:
This is quite funny (and annoying), but after implementing the proper way of resetting the navigation (see [docs](https://reactnavigation.org/docs/7.x/navigation-actions#reset)), I got the same problem that we faced where a reanimated exception would happen when resetting the navigation to a screen that is using the react-navigation header. I figured that if I hide the header when the app is in the background the error doesn't happen. Not sure why when using the old navigation resetting method this problem was not happening.

Reply:
That's a very weird one, but hey, it works.